experimental/InvariantTheory/src/InvariantTheory.jl

```
@@ -0,0 +1,523 @@
+#####################
+#Objects of type ReductiveGroup can be embedded in GLn (for some n) via a representation. This defines the action on a vector space. 
+#We set up an object RepresentationReductiveGroup that carries information about this representation.
+
+mutable struct RepresentationReductiveGroup
+    group::ReductiveGroup
+    rep_mat::MatElem
+    sym_deg::Tuple{Bool, Int}
+    reynolds_v::Function
+    
+    #representation of group G over symmetric degree d
+    function RepresentationReductiveGroup(G::ReductiveGroup, d::Int)
+        R = new()
+        R.group = G
+        R.rep_mat = rep_mat_(G, d)
+        R.sym_deg = (true, d)
+        R.reynolds_v = reynolds_v_slm
+        return R
+    end
+    
+    #matrix M is the representation matrix. does not check M.
+    function RepresentationReductiveGroup(G::ReductiveGroup, M::MatElem)
+        @req base_ring(M) == base_ring(G.group_ideal) "Group ideal and representation matrix must have same parent ring"
+        R = new()
+        R.group = G
+        R.rep_mat = M
+        R.sym_deg = (false, 0)
+        R.reynolds_v = reynolds_v_slm
+        return R
+    end
+end
+
+representation_reductive_group(G::ReductiveGroup, M::MatElem) = RepresentationReductiveGroup(G,M)
+group(R::RepresentationReductiveGroup) = R.group
+
+function representation_on_forms(G::ReductiveGroup, d::Int)
+    @assert G.group[1] == :SL
+        return RepresentationReductiveGroup(G, d)
+end
+
+function representation_reductive_group(G::ReductiveGroup)
+    if G.group[1] == :SL
+        M = canonical_representation(G)
+        return RepresentationReductiveGroup(G,M)
+    end
+end
+
+representation_matrix(R::RepresentationReductiveGroup) = R.rep_mat
+reductive_group(R::RepresentationReductiveGroup) = R.group
+vector_space_dimension(R::RepresentationReductiveGroup) = ncols(R.rep_mat)
+
+function Base.show(io::IO, R::RepresentationReductiveGroup)
+    io = pretty(io)
+    if group(group(R))[1] == :SL
```

Why this restriction?

If you keep it, then raise an `error` in the unhandled cases instead of printing nothing.